presets/cloudflare/httpreq/preset.yaml

```diff
@@ -21,60 +22,164 @@ silver:
+          expr: CASE WHEN array_contains(ARRAY('unknown'), SecurityAction) THEN 'Unknown' WHEN array_contains(ARRAY('allow'), SecurityAction) THEN 'Allowed' WHEN array_contains(ARRAY('block'), SecurityAction) THEN 'Blocked' WHEN array_contains(ARRAY('connectionClose', 'forceConnectionClose'), SecurityAction) THEN 'Dropped' WHEN array_contains(ARRAY('challengeSolved', 'jschallengeSolved', 'managedChallengeNonInteractiveSolved', 'managedChallengeInteractiveSolved'), SecurityAction) THEN 'Approved' WHEN array_contains(ARRAY('skip', 'bypass', 'challengeBypassed', 'jschallengeBypassed', 'managedChallengeBypassed'), SecurityAction) THEN 'No Action' WHEN array_contains(ARRAY('log'), SecurityAction) THEN 'Logged' WHEN array_contains(ARRAY('challenge', 'jschallenge', 'managedChallenge'), SecurityAction) THEN 'Challenge' ELSE 'Other' END
+        - name: disposition_id
+          expr: CASE WHEN array_contains(ARRAY('unknown'), SecurityAction) THEN 0 WHEN array_contains(ARRAY('allow'), SecurityAction) THEN 1 WHEN array_contains(ARRAY('block'), SecurityAction) THEN 2 WHEN array_contains(ARRAY('connectionClose', 'forceConnectionClose'), SecurityAction) THEN 6 WHEN array_contains(ARRAY('challengeSolved', 'jschallengeSolved', 'managedChallengeNonInteractiveSolved', 'managedChallengeInteractiveSolved'), SecurityAction) THEN 8 WHEN array_contains(ARRAY('skip', 'bypass', 'challengeBypassed', 'jschallengeBypassed', 'managedChallengeBypassed'), SecurityAction) THEN 16 WHEN array_contains(ARRAY('log'), SecurityAction) THEN 17 WHEN array_contains(ARRAY('challenge', 'jschallenge', 'managedChallenge'), SecurityAction) THEN 23 ELSE 99 END
+        - name: ocsf_firewall_rule.uid
+          from: SecurityRuleID
+        - name: ocsf_firewall_rule.desc
+          from: SecurityRuleDescription
+        - name: ocsf_firewall_rule.match_location
+          from: WAFMatchedVar
+        - name: ocsf_connection_info.uid
+          from: RayID
+        - name: risk_score
+          expr: CASE WHEN WAFRCEAttackScore >= WAFSQLiAttackScore AND WAFRCEAttackScore >= WAFXSSAttackScore AND WAFRCEAttackScore > 0 THEN WAFRCEAttackScore WHEN WAFSQLiAttackScore >= WAFRCEAttackScore AND WAFSQLiAttackScore >= WAFXSSAttackScore AND WAFSQLiAttackScore > 0 THEN WAFSQLiAttackScore WHEN WAFXSSAttackScore >= WAFRCEAttackScore AND WAFXSSAttackScore >= WAFSQLiAttackScore AND WAFXSSAttackScore > 0 THEN WAFXSSAttackScore ELSE 0 END
+        - name: risk_details
+          expr: CASE WHEN WAFRCEAttackScore >= WAFSQLiAttackScore AND WAFRCEAttackScore >= WAFXSSAttackScore AND WAFRCEAttackScore > 0 THEN 'Remote code execution.' WHEN WAFSQLiAttackScore >= WAFRCEAttackScore AND WAFSQLiAttackScore >= WAFXSSAttackScore AND WAFSQLiAttackScore > 0 THEN 'SQL injection.' WHEN WAFXSSAttackScore >= WAFRCEAttackScore AND WAFXSSAttackScore >= WAFSQLiAttackScore AND WAFXSSAttackScore > 0 THEN 'Cross-site script.' ELSE 'None' END
+        - name: ocsf_unmapped
+          expr: "to_json(
+                  MAP(
```

Comment:
swap this out for a `names_struct` and you can drop all the casting below as it will allow mixed value types.